module/VuFind/src/VuFind/Cover/Router.php

```diff
@@ -85,6 +85,31 @@ public function __construct($url, CoverLoader $coverLoader)
      */
     public function getUrl(RecordDriver $driver, $size = 'small',
         $resolveDynamic = true, $testLoadImage = false
+    ) {
+        $metadata = $this->getMetadata(
+            $driver, $size, $resolveDynamic, $testLoadImage
+        );
+        return $metadata['url'] ?? $metadata;
```

Comment:
Should this be `$metadata['url'] ?? false`? If `$metadata['url']` is unset, then `$metadata` is most likely an empty array, which is not one of the documented return types of this method.

Reply:
It is a bit more complicated - getMetadata from AbstractCover can return empty array, which is then passed from Router::getMetadata to Router::getUrl.

But Router::getMetadata can even return false or null values... we probably want these values to be passed and returned by getUrl I think...

I'll take a look on it and will see...

Reply:
I apologize if I just skipped a step in the code during my review -- that's very possible. If nothing else, it's a sign that we should add a comment here to clarify what is going on, so we don't ask ourselves this question again in the future.

Reply:
I tried to make it more clear and also added a case when empty array is returned from getMetadata, what do you think about it @demiankatz

Reply:
Thanks, @xmorave2. The comment helps to clarify what is going on, but I'm not sure if the implementation matches the comment. If $metadata is `null` or `false`, then `empty($metadata)` will evaluate to true, and the code will normalize the return value to false. If you want to preserve the null return and be sure you never return an array, would it be better to change the `!empty()` check to `!is_array()` instead?

Reply:
You are absolutely right @demiankatz, thanks for catching this. I've fixed it now

Reply:
Great, thank you! I believe we're all set now -- I've just merged this.